force-app/main/default/classes/ProgramEngagementService_TEST.cls

```diff
@@ -0,0 +1,41 @@
+/*
+ *
+ *  * Copyright (c) 2022, salesforce.com, inc.
+ *  * All rights reserved.
+ *  * SPDX-License-Identifier: BSD-3-Clause
+ *  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ *
+ */
+
+@IsTest
+public with sharing class ProgramEngagementService_TEST {
+    private static TestStub bucketSelectorStub;
+
+    private static ProgramEngagementService service = new ProgramEngagementService();
+
+    @IsTest
+    private static void shouldGetActiveStages() {
+        List<String> bucketNames = new List<String>{ 'Active' };
+
+        String stageFieldAPIName = Util.prefixNamespace('Stage__c');
+
+        Schema.SObjectType programEngagementSObjType = ProgramEngagement__c.SObjectType;
+        Schema.SObjectField stageField = Schema.SObjectType.ProgramEngagement__c.fields.getMap()
```

Comment:
I think this is leftover from copy and pasting. Should we be stubbing the selector here?

Reply:
Moving away from the stub was intentional, since the stub implementation (as seen in ServiceDeliveryService_TEST I think) was functionally identical to the method itself I wasn't sure if it was actually adding any value. I wasn't 100% sure, but if your stub is doing exactly what the original method does, why not just test the method as written?

Reply:
The stub is a stand in for the selector so you are not dependent on the selector working in order for your test to pass. It basically is saying I have a selector that should return me these values and when it does then my service should return me these values.

Reply:
I do understand that concept, but if the stub is line-for-line identical to the original selector, I don't think there's any purpose in delivering that behavior via a stub.

Reply:
Thanks for hopping on the call, to recap what we discussed. You are 100% correct we should be using mock data in the return vs replicating the query in the test. This test is fine as is but if you want to take a swing at creating the mock data that'd be cool to have in the repo.